src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const entries = sourceString
+    .split(';')
+    .map((entry) => entry.trim())
+    .filter((entry) => entry);
+
+  const result = {};
+
+  for (const entry of entries) {
+    if (entry.includes(':')) {
```

Comment:
Consider adding an `else` block to handle entries that do not contain a colon. This will ensure that all parts of the input string are processed correctly.